### PR TITLE
Revert #18381 for layout android issues

### PR DESCRIPTION
**Pull request by jeromelaban**
Related to https://github.com/unoplatform/uno/pull/18381, https://github.com/unoplatform/uno/pull/18193 and https://github.com/unoplatform/uno/pull/18377

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/DependencyObjectStore.cs [hunk 1]
```diff
@@ -1327,7 +1327,13 @@ private void CleanupInheritedProperties()
 						return (localProperty, propertyDetails);
 					}
 				}
-				else if (property.IsAttached && property.IsInherited)
+				else if (property.IsAttached
+					&& property.IsInherited
+
+#if __ANDROID__
+					&& _properties.GetPropertyDetails(property) is { }
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/DependencyObjectStore.cs:1334]: Shouldn't this be `FindPropertyDetails`?

File: src/Uno.UI/UI/Xaml/ILayouterElement.cs [hunk 2]
```diff
@@ -74,7 +74,6 @@ internal static bool DoMeasure(this ILayouterElement element, Size availableSize
 			if (isDirty || frameworkElement is null)
 			{
 				// We must reset the flag **BEFORE** doing the actual measure, so the elements are able to re-invalidate themselves
-				// TODO: We are not controlling measure dirty path on Android. If we did in future, we must clear it here as well.
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/ILayouterElement.cs:77]: These comments are still valid

File: src/Uno.UI/UI/Xaml/ILayouterElement.cs [hunk 3]
```diff
@@ -92,9 +91,7 @@ internal static bool DoMeasure(this ILayouterElement element, Size availableSize
 					LayoutInformation.SetAvailableSize(element, availableSize);
 				}
 
-				// TODO: This is NOT correct.
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/ILayouterElement.cs:95]: Same here

File: src/Uno.UI/UI/Xaml/ILayouterElement.cs [hunk 4]
```diff
@@ -113,8 +110,6 @@ internal static bool DoMeasure(this ILayouterElement element, Size availableSize
 				{
 					var previousDesiredSize = childAsUIElement.m_desiredSize;
 					childAsUIElement.EnsureLayoutStorage();
-
-					// TODO: This is NOT correct. This should call DoMeasure (the same method we are in currently!)
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/ILayouterElement.cs:117]: Same here

File: src/Uno.UI/UI/Xaml/ILayouterElement.cs [hunk 5]
```diff
@@ -113,7 +114,7 @@ internal static bool DoMeasure(this ILayouterElement element, Size availableSize
 				{
 					var previousDesiredSize = childAsUIElement.m_desiredSize;
 					childAsUIElement.EnsureLayoutStorage();
-
+
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/ILayouterElement.cs:117]: Likely will be formatting error here

Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/ILayouterElement.cs:117]: Indeed, thanks.

**Comment by jeromelaban** [comment]
@mergifyio backport master

**Pull request merged**